modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2525,32 +2525,36 @@ def binary_op(self, op, right_frame, join_type="outer"):
         PandasDataframe
             New Modin DataFrame.
         """
-        left_parts, right_parts, joined_index, row_lengths = self._copartition(
-            0, right_frame, join_type, sort=True
+        left_parts, list_of_right_parts, joined_index, row_lengths = self._copartition(
+            0, right_frames, join_type, sort=True
         )
         new_left_frame = self.__constructor__(
             left_parts, joined_index, self.columns, row_lengths, self._column_widths
         )
-        new_right_frame = self.__constructor__(
-            right_parts[0],
-            joined_index,
-            right_frame.columns,
-            row_lengths,
-            right_frame._column_widths,
-        )
+        new_right_frames = [
+            self.__constructor__(
+                right_parts,
+                joined_index,
+                right_frames[0].columns,
+                row_lengths,
+                right_frames[0]._column_widths,
```

Comment:
I've missed that earlier... there should be some kind of an assert that all right frames are equally column-ed (both in terms of which columns they have and how they are partitioned), otherwise this piece would break.

Or, probably better, change the loop to something like
```python
new_right_frames = [
    self.__constructor__(
        right_parts,
        joined_index,
        right_frame.columns,
        row_lengths,
        right_frame._column_widths,
    )
    for right_parts, right_frame in zip(list_of_right_parts, right_frames)
]
```

Reply:
Done.